sklearn/datasets/_samples_generator.py

```diff
@@ -99,13 +99,14 @@ def make_classification(n_samples=100, n_features=20, *, n_informative=2,
         classes are balanced. Note that if ``len(weights) == n_classes - 1``,
         then the last class weight is automatically inferred.
         More than ``n_samples`` samples may be returned if the sum of
-        ``weights`` exceeds 1.
+        ``weights`` exceeds 1. Note that the actual class proportions will
+        not exactly match `weights` when `flip_y` isn't 0.
```

Comment:
```suggestion
        ``weights`` exceeds 1. Note that only if ``flip_y=0``, the actual class
    proportions will exactly match ``weights``.
```

Reply:
Honestly I find the original phrasing more natural

Reply:
I tried to avoid the double negative - just a suggestion.